### DISCUSSION TITLE
`suspicious/noExclusiveTests`

### DISCUSSION THREAD

**Discussion by emab** [biomejs/biome#541] [Ideas]
This rule would look for usages where `.only` is used when defining a test. When `.only` is used, it's mostly for debugging purposes and isn't meant to be checked in. If you want to skip tests, they should be explicitly marked with `.skip`.

Example of invalid code:

```js
describe.only("my test", () => {});
it.only("does something", () => {});
```

Example of valid code:

```js
describe.skip("my test", () => {});
it.skip("does something", () => {});
```

An Eslint plugin exists for this rule: https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/docs/rules/no-exclusive-tests.md

**Comment by ematipico** [comment]
Thank you, @emab, for your proposal. I would strongly suggest you add the message at https://github.com/biomejs/biome/discussions/3, where we would like to collect feedback and ideas of rules from other sources.

Also, we usually **don't** assign a group to a rule when it's created. All rules have the group `nursery` set when implemented for the first time. After a trial period, they are promoted to a group, and that's when people can discuss which group fits better.

**Reply by emab** [reply]
Have added this over there, thanks!
